docs/gettingstarted/structuring.md

```diff
@@ -1,54 +1,79 @@
+
+!!! note
+This example is a modified version of the one found in the MDK [here](https://github.com/MinecraftForge/MinecraftForge/blob/1.10.x/mdk/src/main/java/com/example/examplemod/ExampleMod.java)
+
+    package com.examplemod.awesomesauce;
+        
+    import net.minecraft.init.Blocks;
+    import net.minecraftforge.fml.common.Mod;
+    import net.minecraftforge.fml.common.Mod.EventHandler;
+    import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+        
+    @Mod(modid = "awesomesauce", name = "AwesomeSauce", version = "0.1")
+    public class AwesomeSauceMod
+    {
+        @EventHandler
+        public void init(FMLInitializationEvent event)
+        {
+            // some example code
+            System.out.println("DIRT BLOCK >> "+Blocks.DIRT.getUnlocalizedName());
+        }
+    }
 
 What is `@Mod`?
 -------------
 
-This is an annotation indicating to the Forge Mod Loader that the class is a Mod entry point. It contains various metadata about the mod. It also designates the class that will receive `@EventHandler` events. More information can be found at... (Coming Soon)
+`@Mod` is the annotation indicating that the class is a *Mod entry point* for Forge Mod Loader. It contains various bits of Metadata that the mod will use as well as modifiers that tell Forge how to handle your mod. For instance, Forge supports mods written in `Java` and `Scala`, so if you write your mod in `Scala`, you'll need to also set `modLanguage = "scala"` in the `@Mod` annotation so Forge uses the `Scala` compatibility behaviours instead of the default `Java` ones.
+
+Now, the `@Mod` annotation can take quite a few arguments, but the only three *required* ones are `modid`, `name`, and `version`.
+* `modid` is the unique identifier and *internal* name of the mod, and takes a `String`. Usually, this is the same name as the mod, but has some special requirements for the naming convention; namely, it has to be *less than* 64 characters long, and *all lowercase*. Having `modid = "SomeReallyLongAndCamelCasedNameForYourModIDWillNotWorkAndMakeForgeReallyReallyAngry"` is a no-no. It's used internally by Forge and by other Mod Developers that wish to find your mod through Forge (for instance, to use your APIs).
+ * **ONCE YOU PICK THIS IDENTIFIER, YOU SHOULD NOT CHANGE IT DURING THE LIFE OF THE MOD. BAD THINGS HAPPEN IF YOU DO AND OTHER MODS DEPEND ON YOUR MOD.**
+* `name` is the *external* name of the mod, and takes a `String`. This is literally just the name of your mod, in our case it would be `name = "AwesomeSauce"`.
+* `version` is the version of the mod, and takes a `String`. There is no set standard for how you version things, but it's recommended that you read the [versioning](https://mcforge.readthedocs.io/en/latest/conventions/versioning/) article.
+
+Simple, right? `@Mod` has a wide range of other things you can throw at it, like the `modLanguage` example above, but those will be covered in the articles that use them.
```

Comment:
I suggest to replace `modLanguage` with `modLanguageAdapter`, as it probably won't exist anymore in near future (see [here](https://github.com/MinecraftForge/MinecraftForge/pull/4029/files?diff=unified#diff-c8a307019565332befe89f9f61b2975bL161)).